### PR TITLE
amrex: add 23.05

### PR DESCRIPTION
**Pull request by WeiqunZhang**

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256.

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Amrex(CMakePackage, CudaPackage, ROCmPackage):
     mesh refinement (AMR) applications."""
 
     homepage = "https://amrex-codes.github.io/amrex/"
-    url = "https://github.com/AMReX-Codes/amrex/releases/download/23.04/amrex-23.04.tar.gz"
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/amrex/package.py:17]: @WeiqunZhang just a quick note, you don't actually need to update the `url` field in a package when you add a version. Spack will pull out and replace the version internally to generate urls for the new version or older versions.

Comment (WeiqunZhang) [reply var/spack/repos/builtin/packages/amrex/package.py:17]: Got it. Thanks!

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/amrex/package.py:17]: FWIW. I stopped telling ppl that.  Sometimes it's nice to keep the URL up-to-date, especially as older versions are deprecated/removed.

Comment (alecbcs) [reply var/spack/repos/builtin/packages/amrex/package.py:17]: I think that makes sense, it's definitely weird to have the url present for an old version we've deprecated and removed. 

Probably something we could move into another thread, but how'd you like to treat url changes when we automate the checksum verification in CI? If maintainers get into the habit of updating urls that might make it harder to assume old checksums are still valid without retriggering a check of all checksums?

**Pull request merged**